### PR TITLE
[17.0][MIG] account_invoice_refund_link: Migration to 17.0

### PR DESCRIPTION
**Pull request by andrel-exo**

### CODE REVIEW

File: account_invoice_refund_link/__manifest__.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 {
     "name": "Show links between refunds and their originator invoices.",
-    "version": "16.0.1.0.4",
+    "version": "17.0.1.0.4",
```
Comment (MohamedOsman7) [inline account_invoice_refund_link/__manifest__.py]: version should be set at 17.0.1.0.0

**Review by MohamedOsman7 — APPROVED** [review]
Functional review LGTM

File: account_invoice_refund_link/readme/CONTRIBUTORS.rst
Comment (CRogos) [inline account_invoice_refund_link/readme/CONTRIBUTORS.rst:1]: The readme files need to be kept as md files!

**Review by CRogos — CHANGES_REQUESTED** [review]
Please have a look in to the migration guid: 
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

- The commit history is not cleaned up.
- The required commits missing and your changes are mixed up into one commit
   - [IMP] $module: pre-commit auto fixes
   - [MIG] $module: Migration to 17.0

**Review by CRogos — COMMENTED** [review]
Better but not yet 100%. 
According to the migration guide, the last commits should look like this: 
image


Here you can find the instruction how and which commits to squash: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

Here are some samples>
image

**Comment by andrel-exo** [comment]
@CRogos 

Thank you for the support

**Review by CRogos — APPROVED** [review]
Code + Testing LGTM. 
But you please cleanup the commit history? There are a lot of commits by oca-git-bot, ...

File: account_invoice_refund_link/views/account_invoice_view.xml [hunk 2]
```diff
@@ -18,15 +18,15 @@
...
                 >
                     <field name="refund_invoice_ids" nolabel="1">
                         <tree
                             decoration-info="state == 'draft'"
                             decoration-muted="state == 'cancel'"
                         >
-                            <field name="partner_id" invisible="1" />
-                            <field name="move_type" invisible="1" />
+                            <field name="partner_id" column_invisible="True" />
+                            <field name="move_type" column_invisible="True" />
```
Comment (pedrobaeza) [inline account_invoice_refund_link/views/account_invoice_view.xml]: Better to use the usual convention:
```suggestion
                            <field name="partner_id" column_invisible="1" />
                            <field name="move_type" column_invisible="1" />
```

File: account_invoice_refund_link/tests/test_invoice_refund_link.py [hunk 3]
```diff
@@ -8,8 +8,6 @@
 
 
 class TestInvoiceRefundLinkBase(TransactionCase):
-    refund_method = "refund"
```
Comment (pedrobaeza) [inline account_invoice_refund_link/tests/test_invoice_refund_link.py]: Removing this you are doing all the tests the same, and the idea is to test all the scenarios.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: account_invoice_refund_link/views/account_invoice_view.xml (hunk 2 above)
Comment (CRogos) [reply account_invoice_refund_link/views/account_invoice_view.xml]: That was my first thought, too, but when you search the code for column_invisible="1" you get 62 results, and when you search for column_invisible="True" you get 1151 results. 
Of cause for the invisible attribute it is the opposite result ;)

Comment (pedrobaeza) [reply account_invoice_refund_link/views/account_invoice_view.xml]: OK, weird that they decided to go the opposite direction... I think the copy/paste from a non conventional form has done a lot of damage. Anyway, I still think we should stick to `1` in OCA to save chars in the diffs.

Comment (andrel-exo) [reply account_invoice_refund_link/views/account_invoice_view.xml]: Ok, i will change to `1`

File: account_invoice_refund_link/tests/test_invoice_refund_link.py (hunk 3 above)
Comment (andrel-exo) [reply account_invoice_refund_link/tests/test_invoice_refund_link.py]: The `refund_method` field in v17 no longer exists, they use `reverse_moves` method with the parameter `is_modify` to determine what to do with the new credit note. What is the best approach to this situation?

Comment (pedrobaeza) [reply account_invoice_refund_link/tests/test_invoice_refund_link.py]: Then reduce the number of tests, but test both conditions, with and without `is_modify`.

**Review by CRogos — CHANGES_REQUESTED** [review]
could you squash the commits?

image

here is an howto: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

**Comment by CRogos** [comment]
@arleite are you still working on this? The commits need to be cleaned.

**Comment by andrel-exo** [comment]
> @arleite are you still working on this? The commits need to be cleaned.

I hope this is what you wanted, I can't understand the purpose but if it's necessary for the migration, fine. If you need to make any other changes, please suggest them, thanks for your support

**Comment by MohamedOsman7** [comment]
@arleite As @CRogos noted above (see photo with the yellow marker), some commits still can be squshed.  My suggestion:
2024-04-22 16_17_19- 17 0 MIG  account_invoice_refund_link_ Migration to 17 0 by arleite · Pull Req


Here is the documentation:
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

**Review by CRogos — APPROVED** [review]

**Comment by CRogos** [comment]
@pedrobaeza can you merge?

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**